sdk/core/azure-core/src/main/java/com/azure/core/http/policy/BearerTokenAuthenticationPolicy.java

```diff
@@ -36,16 +34,15 @@ public BearerTokenAuthenticationPolicy(TokenCredential credential, String... sco
         Objects.requireNonNull(scopes);
         assert scopes.length > 0;
         this.credential = credential;
-        this.scopes = scopes;
-        this.cache = new SimpleTokenCache(() -> credential.getToken(new TokenRequestContext().addScopes(scopes)));
+        this.tokenRequestContext = new TokenRequestContext().addScopes(scopes);
     }
 
     @Override
     public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
         if ("http".equals(context.getHttpRequest().getUrl().getProtocol())) {
             return Mono.error(new RuntimeException("token credentials require a URL using the HTTPS protocol scheme"));
         }
-        return cache.getToken()
+        return credential.getToken(tokenRequestContext)
```

Comment:
Which custom policies? You mean in service SDKs like Key Vault?

Reply:
This is no longer an issue since this change is reverted for now to keep backward compatibility in azure-core until Azure Identity releases a stable version.